pkg/openSearch/openSearchClient/indexFunctions.go

```diff
@@ -91,129 +85,110 @@ func indexNameSliceOf(resultString []byte) ([]string, error) {
 		return false, errors.WithStack(err)
 	}
 
-	if response.StatusCode == http.StatusNotFound {
-		return false, nil
-	}
-
 	return true, nil
 }
 
 func (i *indexFunction) DeleteIndex(indexName string) error {
-	request := opensearchapi.IndicesDeleteRequest{
-		Index: []string{indexName},
-	}
-
-	response, err := request.Do(context.Background(), i.openSearchProjectClient)
-	if err != nil {
-		return errors.WithStack(err)
-	}
-
-	resultString, err := io.ReadAll(response.Body)
+	_, err := i.openSearchProjectClient.Indices.Delete(
+		context.Background(),
+		opensearchapi.IndicesDeleteReq{
+			Indices: []string{indexName},
+		},
+	)
 	if err != nil {
 		return errors.WithStack(err)
 	}
 
-	return GetResponseError(response.StatusCode, resultString, indexName)
+	return nil
 }
 
 func (i *indexFunction) CreateOrPutAlias(aliasName string, indexNames ...string) error {
-	request := opensearchapi.IndicesPutAliasRequest{
-		Index: indexNames,
-		Name:  aliasName,
-	}
-
-	_, err := request.Do(context.Background(), i.openSearchProjectClient)
+	req, err := i.openSearchProjectClient.Indices.Alias.Put(
+		context.Background(),
+		opensearchapi.AliasPutReq{
+			Indices: indexNames,
+			Alias:   aliasName,
+		},
+	)
 	if err != nil {
-		log.Debug().Msgf("Error while creating and putting alias: %s", err)
 		return errors.WithStack(err)
 	}
 
+	fmt.Printf("%v %v", req, err)
+
 	return nil
 }
 
 func (i *indexFunction) DeleteAliasFromIndex(indexName string, aliasName string) error {
-	request := opensearchapi.IndicesDeleteAliasRequest{
-		Index: []string{indexName},
-		Name:  []string{aliasName},
-	}
-
-	response, err := request.Do(context.Background(), i.openSearchProjectClient)
+	_, err := i.openSearchProjectClient.Indices.Alias.Delete(
+		context.Background(),
+		opensearchapi.AliasDeleteReq{
+			Indices: []string{indexName},
+			Alias:   []string{aliasName},
+		},
+	)
 	if err != nil {
 		return errors.WithStack(err)
 	}
 
-	resultString, err := io.ReadAll(response.Body)
-	if err != nil {
-		return errors.WithStack(err)
-	}
-
-	return GetResponseError(response.StatusCode, resultString, indexName)
+	return nil
 }
 
 func (i *indexFunction) AliasExists(aliasName string) (bool, error) {
-	request := opensearchapi.CatAliasesRequest{
-		Name: []string{aliasName},
-	}
+	req, err := i.openSearchProjectClient.Cat.Aliases(
+		context.Background(),
+		&opensearchapi.CatAliasesReq{
+			Aliases: []string{aliasName},
+		},
+	)
 
-	response, err := request.Do(context.Background(), i.openSearchProjectClient)
 	if err != nil {
 		return false, errors.WithStack(err)
 	}
 
-	if response.StatusCode == http.StatusNotFound {
-		return false, nil
-	}
-	if response.Header.Get("Content-Length") == `0` {
-		log.Debug().Msgf("Alias %s does not exist", aliasName)
-		return false, nil
+	if len(req.Aliases) == 0 {
+		log.Debug().Str("src", "opensearch").Msgf("alias %s does not exist", aliasName)
+		return false, errors.WithStack(err)
```

Comment:
there was no error returned if the alias did not exist

Reply:
Fixed in 1ef6a9bb9ced91f0556b40884cee43f80fde89d6, thanks!

Reply:
I think, there is a misunderstanding. On main, there will be no error (nil) returned if the alias does not exist. https://github.com/greenbone/opensight-golang-libraries/blob/main/pkg/openSearch/openSearchClient/indexFunctions.go#L179

This behavior should not be changed, so instead of returning an error, we should return nil, if the alias does not exist.